addons/hr_holidays/models/hr_leave_accrual_plan.py

```diff
@@ -137,6 +157,30 @@ def action_open_accrual_plan_employees(self):
             'domain': [('id', 'in', self.allocation_ids.employee_id.ids)],
         }
 
+    def action_create_accrual_plan_level(self):
+        action = self.env['ir.actions.act_window']._for_xml_id('hr_holidays.action_create_accrual_plan_level')
+        action['context'] = {
+                'new': True,
+                'is_carryover': self.is_carryover,
+                'accrued_gain_time': self.accrued_gain_time,
+                'can_modify_value_type': not self.time_off_type_id and not self.level_ids,
+                'added_value_type': self.added_value_type,
+            }
+        return action
+
+
+    def action_open_accrual_plan_level(self,level_id):
+        action = self.env['ir.actions.act_window']._for_xml_id('hr_holidays.action_create_accrual_plan_level')
+        action['name'] = _('Milestone Edition')
+        action['res_id'] = level_id
+        return action
+
+    @api.constrains('is_carryover')
+    def constrain_action_with_unused_accruals_on_levels(self):
+        for plan in self.filtered(lambda p: not p.is_carryover):
+            for level in plan.level_ids:
+                level.action_with_unused_accruals = "lost"
```

Comment:
```suggestion
        for plan in self:
            if p.is_carryover:
                continue
            for level in plan.level_ids:
                level.action_with_unused_accruals = "lost"
```
to avoid a double loop :)